tests/integration/test_integration.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = METADATA["name"]
+
```

Comment:
We probably want to add `mongodb-k8s` to the picture and have a test that brings the charm to active/idle

Reply:
I'm adding `mongodb-k8s`, however, we cannot bring the charm to active idle yet, as we don't have the nrf provider side implemented.

Reply:
I've set the itest relying on mongo to xfail. I can create an issue. The crash is related with some lib used in the `tls-certificates-interface`
```logs
unit-sdcore-smf-operator-0: 2023-05-19 12:44:33 DEBUG juju.worker.uniter.storage next hook op for storage-config-0: {Kind:2 Life:alive Attached:true Location:/var/lib/juju/storage/config/0}
unit-sdcore-smf-operator-0: 2023-05-19 12:44:33 DEBUG juju.worker.uniter no operations in progress; waiting for changes
unit-mongodb-k8s-0: 2023-05-19 12:44:34 WARNING unit.mongodb-k8s/0.install Traceback (most recent call last):
unit-mongodb-k8s-0: 2023-05-19 12:44:34 WARNING unit.mongodb-k8s/0.install   File "./src/charm.py", line 34, in <module>
unit-mongodb-k8s-0: 2023-05-19 12:44:34 WARNING unit.mongodb-k8s/0.install     from charms.mongodb.v0.mongodb_tls import MongoDBTLS
unit-mongodb-k8s-0: 2023-05-19 12:44:34 WARNING unit.mongodb-k8s/0.install   File "/var/lib/juju/agents/unit-mongodb-k8s-0/charm/lib/charms/mongodb/v0/mongodb_tls.py", line 16, in <module>
unit-mongodb-k8s-0: 2023-05-19 12:44:34 WARNING unit.mongodb-k8s/0.install     from charms.tls_certificates_interface.v1.tls_certificates import (
unit-mongodb-k8s-0: 2023-05-19 12:44:34 WARNING unit.mongodb-k8s/0.install   File "/var/lib/juju/agents/unit-mongodb-k8s-0/charm/lib/charms/tls_certificates_interface/v1/tls_certificates.py", line 257, in <module>
unit-mongodb-k8s-0: 2023-05-19 12:44:34 WARNING unit.mongodb-k8s/0.install     from cryptography import x509
unit-mongodb-k8s-0: 2023-05-19 12:44:34 WARNING unit.mongodb-k8s/0.install   File "/var/lib/juju/agents/unit-mongodb-k8s-0/charm/venv/cryptography/x509/__init__.py", line 6, in <module>
unit-mongodb-k8s-0: 2023-05-19 12:44:34 WARNING unit.mongodb-k8s/0.install     from cryptography.x509 import certificate_transparency
unit-mongodb-k8s-0: 2023-05-19 12:44:34 WARNING unit.mongodb-k8s/0.install   File "/var/lib/juju/agents/unit-mongodb-k8s-0/charm/venv/cryptography/x509/certificate_transparency.py", line 10, in <module>
unit-mongodb-k8s-0: 2023-05-19 12:44:34 WARNING unit.mongodb-k8s/0.install     from cryptography.hazmat.bindings._rust import x509 as rust_x509
unit-mongodb-k8s-0: 2023-05-19 12:44:34 WARNING unit.mongodb-k8s/0.install ImportError: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /var/lib/juju/agents/unit-mongodb-k8s-0/charm/venv/cryptography/hazmat/bindings/_rust.abi3.so)
unit-mongodb-k8s-0: 2023-05-19 12:44:34 ERROR juju.worker.uniter.operation hook "install" (via hook dispatching script: dispatch) failed: exit status 1
unit-mongodb-k8s-0: 2023-05-19 12:44:34 DEBUG juju.machinelock created rotating log file "/var/log/machine-lock.log" with max size 10 MB and max backups 5
unit-mongodb-k8s-0: 2023-05-19 12:44:34 DEBUG juju.machinelock machine lock "machine-lock" released for mongodb-k8s/0 uniter (run install hook)
unit-mongodb-k8s-0: 2023-05-19 12:44:34 DEBUG juju.worker.uniter.operation lock released for mongodb-k8s/0
unit-mongodb-k8s-0: 2023-05-19 12:44:34 INFO juju.worker.uniter awaiting error resolution for "install" hook
unit-mongodb-k8s-0: 2023-05-19 12:44:34 DEBUG juju.worker.uniter [AGENT-STATUS] error: hook failed: "install"
unit-mongodb-k8s-0: 2023-05-19 12:44:34 DEBUG juju.worker.uniter.remotestate storage attachment change for mongodb-k8s/0: {storage-db-1 {2 alive true /data/db}}
unit-mongodb-k8s-0: 2023-05-19 12:44:34 INFO juju.worker.uniter awaiting error resolution for "install" hook
model-72ff20dc-7a57-45fb-8871-87ea73330d60: 2023-05-19 12:44:34 DEBUG juju.worker.caasadmission received admission request for sdcore-smf-operator-0 of /v1, Kind=Pod in namespace test-integration-gxwm
controller-0: 2023-05-19 12:44:34 DEBUG juju.worker.caasapplicationprovisioner.sdcore-smf-operator updating application "sdcore-smf-operator" trust to false
model-72ff20dc-7a57-45fb-8871-87ea73330d60: 2023-05-19 12:44:34 DEBUG juju.worker.caasadmission received admission request for sdcore-smf-operator of rbac.authorization.k8s.io/v1, Kind=Role in namespace test-integration-gxwm
model-72ff20dc-7a57-45fb-8871-87ea73330d60: 2023-05-19 12:44:34 DEBUG juju.worker.caasadmission received admission request for test-integration-gxwm-sdcore-smf-operator of rbac.authorization.k8s.io/v1, Kind=ClusterRole in namespace 
unit-sdcore-smf-operator-0: 2023-05-19 12:44:36 DEBUG juju.worker.uniter.remotestate storage attachment change for sdcore-smf-operator/0: {storage-config-0 {2 alive true /var/lib/juju/storage/config/0}}
unit-sdcore-smf-operator-0: 2023-05-19 12:44:36 DEBUG juju.worker.uniter.storage next hook op for storage-config-0: {Kind:2 Life:alive Attached:true Location:/var/lib/juju/storage/config/0}
unit-sdcore-smf-operator-0: 2023-05-19 12:44:36 DEBUG juju.worker.uniter no operations in progress; waiting for changes
```

Reply:
Deploying and relating to mongodb shouldn't be an "expected failed". We expect this to work.

Reply:
Fixed. I've already added the nrf, we're waiting for this [PR](https://github.com/canonical/sdcore-nrf-operator/pull/12)